### PR TITLE
release-notes: added release notes for v4.5.0

### PR DESCRIPTION
**Pull request by hilmarm**
Signed-off-by: Hilmar Magnusson <hilmar.magnusson@bisdn.de>

### CODE REVIEW

File: release-notes.txt [hunk 1]
```diff
@@ -0,0 +1,235 @@
+BISDN Linux
+v4.5.0 Release Notes
+Date: 2022-03-15
+
+! New in this release
+- baseboxd:
+  - updated version to 1.11.1
+  - added initial support for Broadcom KNET interfaces.
+- Linux kernel updated to version 5.15.22
+- iproute2: udpate to version 5.16
```
Comment (jklare) [inline release-notes.txt:10]: i think we also added support for STP on bonded interfaces (https://github.com/bisdn/bisdn.github.io/pull/136)

Comment (jklare) [reply release-notes.txt:10]: and we fixed this bug here: https://github.com/bisdn/basebox/issues/364

Comment (hilmarm) [reply release-notes.txt:10]: Mentioned STP support on bonded interfaces, and added the bug fix to list of bugfixes

File: release-notes.txt [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+
+! Experimental features
+- OF-DPA: initial support for creating and enabling Broadcom KNET interfaces.
+
+! Bug fixes
+- fixed bug related to terminaton MAC entries not being correctly removed on IP deletion.
+- resolved STP issue where the initial STP state was missed, causing ports to forward/flood instead of blocking.
+- fixed an issue in where any reachable ip neighbor was assumed to be routable even if a route did not exist.
```
Comment (jklare) [inline release-notes.txt:19]:
```suggestion
- fixed an issue where any reachable ip neighbor was assumed to be routable even if a route did not exist.
```

File: release-notes.txt [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+  - updated version to 1.11.1
+  - added support for Spanning Tree Protocol (STP, RSTP and MSTP) on bonded interfaces.
+  - added initial support for Broadcom KNET interfaces.
+- Linux kernel updated to version 5.15.22
+- iproute2: udpate to version 5.16
+
+! Experimental features
+- OF-DPA: initial support for creating and enabling Broadcom KNET interfaces.
+
+! Bug fixes
```
Comment (jklare) [inline release-notes.txt:16]: i think linking to the issues created (and now resolved) on github would be helpful here

Comment (hilmarm) [reply release-notes.txt:16]: Linked to github issues/mr where it made sense

File: release-notes.txt [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+- fixed bug related to terminaton MAC entries not being correctly removed on IP deletion. (https://github.com/bisdn/basebox/issues/362)
+- resolved STP issue where the initial STP state was missed, causing ports to forward/flood instead of blocking. (https://github.com/bisdn/bisdn.github.io/pull/136)
+- fixed an issue where any reachable ip neighbor was assumed to be routable even if a route did not exist. (https://github.com/bisdn/basebox/issues/364)
+- ofdpa: fixed 1G ports on PHY2/3 on platform as4630-54pe.
```
Comment (jklare) [inline release-notes.txt:20]: what was fixed here?

File: release-notes.txt (hunk 3 above)
Comment (jklare) [reply release-notes.txt:16]: lgtm

File: release-notes.txt [hunk 5]
```diff
@@ -0,0 +1,237 @@
...
+  - added initial support for Broadcom KNET interfaces.
+- Linux kernel updated to version 5.15.22
+- iproute2: udpate to version 5.16
+
+! Experimental features
+- OF-DPA: initial support for creating and enabling Broadcom KNET interfaces.
+
+! Bug fixes
+- fixed bug related to terminaton MAC entries not being correctly removed on IP deletion. (https://github.com/bisdn/basebox/issues/362)
+- resolved STP issue where the initial STP state was missed, causing ports to forward/flood instead of blocking. (https://github.com/bisdn/bisdn.github.io/pull/136)
```
Comment (jklare) [inline release-notes.txt:18]: i think the link does not fit here

File: release-notes.txt (hunk 4 above)
Comment (hilmarm) [reply release-notes.txt:20]: This was the polarity flip fix, the commit message reads:

> The polarity flip needs to be set for the 1g ports on PHY2/PHY3 to work, so set it for all 1G ports.

Comment (KanjiMonster) [reply release-notes.txt:20]:
```suggestion
- ES202 (AS4630-54PE): fixed an issue preventing some of the 1G ports from transmitting packets
```

File: release-notes.txt (hunk 5 above)
Comment (hilmarm) [reply release-notes.txt:18]: Updated to correct pull request in basebox

File: release-notes.txt [hunk 6]
```diff
@@ -0,0 +1,237 @@
...
+! New in this release
+- baseboxd:
+  - updated version to 1.11.1
+  - added support for Spanning Tree Protocol (STP, RSTP and MSTP) on bonded interfaces.
+  - added initial support for Broadcom KNET interfaces.
+- Linux kernel updated to version 5.15.22
+- iproute2: udpate to version 5.16
+
+! Experimental features
+- OF-DPA: initial support for creating and enabling Broadcom KNET interfaces.
```
[the remark this answers is not in the document]
Comment (hilmarm) [reply release-notes.txt:14]: I added both projects

File: release-notes.txt [hunk 7]
```diff
@@ -0,0 +1,237 @@
+BISDN Linux
+v4.5.0 Release Notes
+Date: 2022-03-15
+
+! New in this release
+- baseboxd:
+  - updated version to 1.11.1
+  - added support for Spanning Tree Protocol (STP, RSTP and MSTP) on bonded interfaces
```
Comment (KanjiMonster) [inline release-notes.txt:8]: This was already part of 4.4.0 (https://github.com/bisdn/bisdn-linux/blob/v4.4.0/release-notes.txt#L10)

File: release-notes.txt [hunk 8]
```diff
@@ -0,0 +1,237 @@
...
+  - added initial support for Broadcom KNET interfaces
+- Linux kernel updated to version 5.15.22
+- iproute2: udpated to version 5.16
```
Comment (KanjiMonster) [inline release-notes.txt:11]: might be worthwhile to add a one liner as ! Changed behaviour that you now need to use `bridge -d vlan` instead of `bridge vlan database` to see per-vlan stp states.

File: release-notes.txt (hunk 7 above)
Comment (hilmarm) [reply release-notes.txt:8]: Our testing pipelines however were allowed to fail for bonded STP up until two weeks ago (when we no longer saw failing bonded stp tests), should we reword it to say 'now stable support added'?

File: release-notes.txt (hunk 8 above)
Comment (hilmarm) [reply release-notes.txt:11]: Added section on changed behaviour

File: release-notes.txt (hunk 7 above)
Comment (KanjiMonster) [reply release-notes.txt:8]: It feels weird having that tied to the list of baseboxd changes, since this isn't a baseboxd change.

I would maybe add a line to the New in this release along side
```
- stp on bond interfaces is now fully supported and considered stable
```

Do we consider it stable? @jklare

File: release-notes.txt [hunk 9]
```diff
@@ -0,0 +1,241 @@
+BISDN Linux
+v4.5.0 Release Notes
+Date: 2022-03-15
+
+! New in this release
+- baseboxd:
+  - updated version to 1.11.1
+  - added initial support for Broadcom KNET interfaces
```
Comment (KanjiMonster) [inline release-notes.txt:8]: Since this is now basically duplicated in the experimental features, we might was well drop this and compress this to one line:
```suggestion
- baseboxd: updated to version 1.11.1
```

File: release-notes.txt [hunk 10]
```diff
@@ -0,0 +1,241 @@
...
+- Linux kernel updated to version 5.15.22
+- iproute2: udpated to version 5.16
+- stp on bond interfaces is now fully supported and considered stable
+
+
+! Experimental features
+- basebox/OF-DPA: initial support added for creating and enabling Broadcom KNET interfaces.
```
Comment (KanjiMonster) [inline release-notes.txt:15]:
```suggestion
- baseboxd/OF-DPA: initial support added for creating and enabling Broadcom KNET interfaces.
```

**Review by KanjiMonster — APPROVED** [review]
